frontend/src/Pages/OpenContentLevelDashboard.tsx

```diff
@@ -29,6 +30,21 @@ export default function OpenContentLevelDashboard() {
             navigate(`/knowledge-center-management/libraries`);
         }
     }
+    async function handleHelpfulLinkClick(id: number): Promise<void> {
+        const response = await API.put<{ url: string }, null>(
+            `/helpful-links/activity/${id}`,
+            null
+        );
+        if (response.success) {
+            if (Array.isArray(response.data)) {
```

Comment:
This can be fixed/made more concise by:
```ts
const resp = await API.put<{url: string}>(`helpful-links/activity/${id}`, null) as ServerResponseOne<{url: string}>;
if (response.success) {
    window.open(resp.data?.url, '_blank');
}
```